src/gui/controllers/SensorDataFeatureController.java

```diff
@@ -0,0 +1,48 @@
+package gui.controllers;
+
+import javafx.fxml.FXML;
+import javafx.scene.text.Text;
+import remote.Car;
+import remote.listeners.DataListener;
+
+/**
+ * Sensor Data Grid-feature controller. This is the main controller class for the Sensor Data Grid-feature.
+ * @author Henrik Nilsson
+ */
+public class SensorDataFeatureController {
+
+    @FXML private Text sensorDataAccelerationX;
+    @FXML private Text sensorDataAccelerationY;
+    @FXML private Text sensorDataAccelerationZ;
+    @FXML private Text sensorDataSpeed;
+    @FXML private Text sensorDataDistance;
+
+    /**
+     * Initialize method called by JavaFX thread when JavaFX-objects are constructed and ready.
+     * This method sets up listeners, styles the input dialog and adds formatting to the input dialog.
+     */
+    public void initialize() {
+        // Add listeners
+        DataListener<Integer> accXListener = (Integer data) -> {
+            sensorDataAccelerationX.setText(String.valueOf(data));
+        };
+        DataListener<Integer> accYListener = (Integer data) -> {
+            sensorDataAccelerationY.setText(String.valueOf(data));
+        };
+        DataListener<Integer> accZListener = (Integer data) -> {
+            sensorDataAccelerationZ.setText(String.valueOf(data));
+        };
+        DataListener<Integer> speedListener = (Integer data) -> {
+            sensorDataSpeed.setText(String.valueOf(data));
+        };
+        DataListener<Integer> distanceListener = (Integer data) -> {
+            sensorDataDistance.setText(String.valueOf(data));
+        };

```

Comment:
In the requirements specification we have said that the remote model will show the sensor data in SI units, are these values scaled correctly?

Reply:
Jag har tittat närmare på vilka enheter sensorerna använda. För accelerometerdatan är enheten mg, alltså tusendelar av tyngdaccelerationen. Möjiga värden är i intervallet +-2 g. Hastigheten är i dm/s med upplösningen 1 dm/s. Avståndet är i millimeter och är tillförlitligt upp till åtminstone 700 mm, 1500 mm under optimala förhållanden.

Reply:
Denna feature skriver ut rådata och gör ingen egen formatering